spring-beans/src/main/java/org/springframework/beans/BeanUtils.java

```diff
@@ -710,17 +713,25 @@ private static void copyProperties(Object source, Object target, @Nullable Class
 			if (writeMethod != null && (ignoreList == null || !ignoreList.contains(targetPd.getName()))) {
 				PropertyDescriptor sourcePd = getPropertyDescriptor(source.getClass(), targetPd.getName());
 				if (sourcePd != null) {
+					Field sourcefield = ReflectionUtils.findField(source.getClass(), sourcePd.getName());
+					Field targetfield = ReflectionUtils.findField(target.getClass(), targetPd.getName());
+										
+					TypeDescriptor sourceTypeDescriptor = new TypeDescriptor(sourcefield);
+					TypeDescriptor targetTypeDescriptor = new TypeDescriptor(targetfield);
+					
 					Method readMethod = sourcePd.getReadMethod();
+					
 					if (readMethod != null &&
-							ClassUtils.isAssignable(writeMethod.getParameterTypes()[0], readMethod.getReturnType())) {
+							ClassUtils.isAssignable(writeMethod.getParameterTypes()[0], readMethod.getReturnType()) &&
+							sourceTypeDescriptor.getResolvableType().equals(targetTypeDescriptor.getResolvableType())) {
 						try {
 							if (!Modifier.isPublic(readMethod.getDeclaringClass().getModifiers())) {
 								readMethod.setAccessible(true);
 							}
 							Object value = readMethod.invoke(source);
 							if (!Modifier.isPublic(writeMethod.getDeclaringClass().getModifiers())) {
 								writeMethod.setAccessible(true);
-							}
+							}							
```

Comment:
Please make sure to run `./gradlew check` before submitting PRs in order to avoid Checkstyle violations that break the build.

Reply:
Yes. I will make sure to run `./gradlew check`  before committing. Thanks for your guidance